goldstone/core/serializers.py

```diff
@@ -98,3 +98,95 @@ def to_representation(self, instance):
+        and add the metadata to the return result. We chose the latter.
+
+        :type instance: Result
+        :param instance: One instance from an Elasticsearch search response
+        :rtype: dict
+        :return: The response minus exclusions, plus some metadata values
+
+        """
+
+        # These metadata fields will be added to the return value.
+        METADATA = ["doc_type", "id", "index"]
+
+        # Get the standard to_dict() result...
+        result = super(EventSerializer, self).to_representation(instance)
+
+        # ...now add non-None/bank metadata fields and values to it.
+        for field in METADATA:
+            if instance.meta.get(field):
+                result[field] = instance.meta[field]
+
+        return result
+
+
+class EventAggSerializer(ReadOnlyElasticSerializer):
+    """Custom serializer for the aggregation that's returned from
+    Elasticsearch.
+
+    Copied from LogEventAggSerializer, because LEAS is scheduled for deletion.
+
+    """
+
+    def to_representation(self, instance):
+        """Create serialized representation of aggregate log data.
+
+        There will be a summary block that can be used for ranging, legends,
+        etc., then the detailed aggregation data which will be a nested
+        structure.  The number of layers will depend on whether the host
+        aggregation was done.
+
+        """
+
+        timestamps = [i['key'] for i in instance.per_interval['buckets']]
+        event_types = [i['key'] for i in instance.per_type['buckets']]
+        hosts = [i['key'] for i in instance.per_host['buckets']] \
+            if hasattr(instance, 'per_host') else None
+
+        # let's clean up the inner buckets
+        data = []
+
+        if hosts is None:
+            for interval_bucket in instance.per_interval.buckets:
+                key = interval_bucket.key
+                values = [{item.key: item.doc_count}
+                          for item in interval_bucket.per_type.buckets]
+                data.append({key: values})
+
+        else:
+            for interval_bucket in instance.per_interval.buckets:
+                interval_key = interval_bucket.key
+                interval_values = []
+                for host_bucket in interval_bucket.per_host.buckets:
+                    key = host_bucket.key
+                    values = [{item.key: item.doc_count}
+                              for item in host_bucket.per_type.buckets]
+                    interval_values.append({key: values})
+                data.append({interval_key: interval_values})
+
+        if hosts is None:
+            return {'timestamps': timestamps,
+                    'types': event_types,
+                    'data': data
+                    }
+        else:
+            return {'timestamps': timestamps,
+                    'hosts': hosts,
+                    'types': event_types,
+                    'data': data
+                    }
```

Comment:
Probably will need to do some sort of localized bucketing (host or id or something) when we go at events from the perspective of a node on the graph rather than globally.

John

jxstanford@gmail.com
@jxstanford



> On Jun 12, 2015, at 13:04:52, John DeRosa <notifications@github.com> wrote:
> 
> In goldstone/core/serializers.py <https://github.com/Solinea/goldstone-server/pull/47#discussion_r32349911>:
> 
> > +                    values = [{item.key: item.doc_count}
> > +                              for item in host_bucket.per_type.buckets]
> > +                    interval_values.append({key: values})
> > +                data.append({interval_key: interval_values})
> > +
> > +        if hosts is None:
> > +            return {'timestamps': timestamps,
> > +                    'types': event_types,
> > +                    'data': data
> > +                    }
> > +        else:
> > +            return {'timestamps': timestamps,
> > +                    'hosts': hosts,
> > +                    'types': event_types,
> > +                    'data': data
> > +                    }
> @jxstanford <https://github.com/jxstanford>: Hmm, I could rip out the check for hosts from this, as it's used now. Will it ever make sense to want to bucketize the events based on host?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Solinea/goldstone-server/pull/47/files#r32349911>.
>